### ISSUE TITLE
How to achieve good performance

### ISSUE THREAD

**Issue by JasonSloan** [ultralytics/ultralytics#19812]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi!

If I try to train yolo11s-detect from scratch using coco2017 datasets with your default hyperparameters, can I get  roughly the same performance as yours? Do you use extra training dataset to improve performance? 

I want to know this because I got a poor performance in a seneriao person detection task. After I finetuned for 200 epochs, the mAP was much more low than your pretrained model could acheive. The training dataset was very small, only about 4000 images, and they come from 4 fixed background with only different persons or same persons in different time. After I trained the model, I tested it in other 3 fixed background labeled images and come to a conclusion mentioned before.

I am really eager to know why would this happen? Can you help me figure out about this? Or can you give me some guidelines?

Many thanks!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @JasonSloan, thank you for your interest in Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com/) for helpful resources, including detailed guides on [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage. Many common questions and training tips are addressed there.

If this is a custom training ❓ question, here are some steps to help us assist you better:
- Provide more details about your dataset, such as class distribution, image resolution, or any preprocessing steps.
- Share relevant training logs, including loss metrics and validation results, if possible.
- Verify you're following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/), which include advice on hyperparameters, augmentation strategies, and data preparation.

If this is a 🐛 bug report, we kindly ask for a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us identify and resolve the issue.

A few key notes for your scenario:
1. Using a small dataset (e.g., 4000 images) may result in overfitting, especially when training on a fixed background. Consider diversifying your dataset with more varied samples or augmentations to improve generalization.
2. Fine-tuning for 200 epochs could potentially overfit the model to your small dataset. Experiment with fewer epochs or use early stopping to prevent this.
3. Pretrained weights often yield better results on limited data compared to training from scratch. Consider fine-tuning a model with pretrained weights for better performance.

For community interaction and real-time support, feel free to join us on [Discord](https://discord.com/invite/ultralytics) 🎧, participate in discussions on [Discourse](https://community.ultralytics.com/), or share insights on our [Subreddit](https://reddit.com/r/Ultralytics).

## Upgrade

To ensure you're using the latest features and fixes, upgrade the `ultralytics` package (along with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml)) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be utilized in the following verified environments:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing. These tests ensure correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across macOS, Windows, and Ubuntu.

This is an automated response to guide you in the right direction. An Ultralytics engineer will also review your issue and provide further assistance soon. Thank you for your patience and for being a part of our community! 🌟

**Comment by glenn-jocher** [comment]
@JasonSloan hi! For optimal results with small datasets (like your 4k images), we recommend:  
1. **Use pretrained weights** instead of training from scratch to leverage learned features.  
2. **Augment your data** (rotate, flip, adjust lighting) to improve generalization beyond fixed backgrounds.  
3. **Adjust hyperparameters**: Start with higher learning rates (`warmup_epochs=0`) and extend training (200+ epochs).  

Your lower mAP is likely due to limited dataset diversity/size. See our [Fine-Tuning YOLO11 Models guide](https://docs.ultralytics.com/guides/model-evaluation-insights/#how-can-i-fine-tune-a-pre-trained-yolo11-model-for-my-specific-dataset) for detailed strategies. For community insights, join the [Ultralytics Discord](https://discord.com/invite/ultralytics). 😊
